src/emulator/storage/rules/manager.ts

```diff
@@ -1,52 +1,72 @@
 import * as chokidar from "chokidar";
-import * as fs from "fs";
 import { EmulatorLogger } from "../../emulatorLogger";
 import { Emulators } from "../../types";
-import { FirebaseError } from "../../../error";
 import { SourceFile } from "./types";
 import { StorageRulesIssues, StorageRulesRuntime, StorageRulesetInstance } from "./runtime";
+import { RulesConfig } from "..";
 
 /**
- * Loads and maintains a {@link StorageRulesetInstance} for a given source file. Listens for
- * changes to the file and updates the ruleset accordingly.
+ * Keeps track of the rules source file and maintains a generated ruleset for one or more storage
+ * resources.
+ * */
+export interface StorageRulesManager {
+  /** Sets source file for each resource using the rules previously passed in the constructor. */
+  start: () => Promise<StorageRulesIssues>;
+
+  /** Retrieves the generated ruleset for the resource. */
+  getRuleset: (resource: string) => StorageRulesetInstance | undefined;
+
+  /**
+   * Updates the source file and, correspondingly, the file watcher and ruleset for the resource.
+   */
+  setSourceFile: (rules: SourceFile, resource: string) => Promise<StorageRulesIssues>;
```

Comment:
I'm a little confused why you would want to have two ways of passing a `SourceFile` to the `StorageRulesManager`. IMO it's unclear when you would use the constructor (mentioned in L13) vs `setSourceFile`

Reply:
Right, for the leaf class it's sort of the same thing. But for the registry `start` is like a `setAllSourceFile`.

I see what you mean about the usage being unclear though. Would renaming help? e.g. `updateSourceFile`. Or do you want to change the methods?